### DISCUSSION TITLE
How to use Custom Asset Metadata to filter a data pool using the new Pimcore Query Language?

### DISCUSSION THREAD

**Discussion by DanielLaranjeira063** [pimcore/pimcore#17652] [Q&A / Support] [answered]
Say I have an Asset metadata class definition with the following structure:
General Settings:
- Name: Generic
- Prefix: Generic
-- Panel:
--- Name: Generic
--- Title: Generic
---- Select input
----- Name: imagestatus
----- Title: Image Status
----- Options: Active/Inactive

How would I use PQL to filter for this condition in the pre condition servive in Portal Engine?

Print screen:
My asset metadata class definition:
![image](https://github.com/user-attachments/assets/6ddb0857-1801-4514-a406-ea57a66bb89a)

My image custom metadata:
![image](https://github.com/user-attachments/assets/09aa1b94-0107-4b39-b5e1-aaa3696335e1)
![image](https://github.com/user-attachments/assets/0dd9d6b1-b5a9-4437-9281-ab531cba7147)

PQL:
![image](https://github.com/user-attachments/assets/6f3bcc1a-0541-4b18-a74c-1f55e00852af)

Result:
![image](https://github.com/user-attachments/assets/31a8ee70-93b3-4751-9fef-b5d3065ad727)

I can't figure for the life of me on how to get this working. Any ideas?

**Comment by kingjia90** [comment]
just `imagestatus` should work, `generic` is a layout thing in there

**Reply by DanielLaranjeira063** [reply]
> just `imagestatus` should work, `generic` is a layout thing in there

![image](https://github.com/user-attachments/assets/e9312985-db56-49c7-ad7d-5720563caafe)

Should work, but it does not

**Comment by DanielLaranjeira063** [comment, marked as answer]
Had to search through the open search indexes and do a little trial and error, but finally found out. 
The exact query is: custom_fields.PortalEngineBundle.standard_fields.Generic.imagestatus="Active"
Feels "hacky" and too long, there should be a better way to do this, but for now it works
